framework/test/mocha/common/utils/blocks_and_transactions_helper.js

```diff
@@ -0,0 +1,250 @@
+/*
+ * Copyright © 2019 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
+const util = require('util');
+const Promise = require('bluebird');
+const { transfer } = require('@liskhq/lisk-transactions');
+const Bignum = require('../../../../src/modules/chain/helpers/bignum');
+const random = require('../../common/utils/random');
+const localCommon = require('../../integration/common');
+const BlockLogic = require('../../../../src/modules/chain/logic/block');
+const accountFixtures = require('../../fixtures/accounts');
+const slots = require('../../../../src/modules/chain/helpers/slots');
+const {
+	registeredTransactions,
+} = require('../../common/registered_transactions');
+const InitTransaction = require('../../../../src/modules/chain/logic/init_transaction');
+
+const initTransaction = new InitTransaction({ registeredTransactions });
+const { NORMALIZER } = global.__testContext.config;
+const addTransaction = util.promisify(localCommon.addTransaction);
+const forge = util.promisify(localCommon.forge);
+const isTransactionInPool = localCommon.transactionInPool;
+
+const addTransactionsAndForge = util.promisify(
+	localCommon.addTransactionsAndForge
+);
+
+const TYPE = {
+	RECEIVED: createCreditTransaction, // CREDIT
+	SPEND: createDebitTransaction, // DEBIT
+};
+
+const EXPECT = {
+	OK: true,
+	FAIL: false,
+};
+
+function createDebitTransaction(account, amount) {
+	return transfer({
+		amount: new Bignum(NORMALIZER).multipliedBy(amount).toString(),
+		recipientId: random.account().address,
+		passphrase: account.passphrase,
+	});
+}
+
+function createCreditTransaction(account, amount) {
```

Comment:
don't we have this function somewhere in commons?

Reply:
Yes, I think so too. We have most of these functions already declared elsewhere. Adding this class does not make sense to me. It duplicates code and adds technical debt. I'd prefer if we reused existing functions.

Reply:
We used same existing code, just wrapped it to write better and quick tests scenarios. So I don't think its something of a problem here. If we had re-invented the `transfer` again and have to maintain on two places then it was a technical debt. 


@4miners

Reply:
I also kind of feel like this BAT class needs its own unit tests. Isn't it too complex for a helper? I'm not sure if we can get rid of it in 2.0 but maybe in upcoming versions we should avoid having helper classes of helpers :D

Reply:
@yatki If you look at many other tests you will see its not too complex. I agree that tests and expectations can be more declarative, but we can do it as separate issue.

Reply:
Created a separate issue to refactor these tests. Please share your thoughts to it.